src/Shouldly.Tests/ShouldBeEquivalentTo/ObjectScenario.cs

```diff
@@ -285,5 +287,15 @@ public void ShouldPassWhenObjectContainsInfiniteLoop()
 
             subject.ShouldBeEquivalentTo(expected);
         }
+
+        [Fact]
+        public void ShouldThrowSensibleErrorWhenIndexersUsed()
+        {
+            var subject = new IndexableObject(new List<string>{"foo", "bar"});
+            var expected = new IndexableObject(new List<string>{"a", "b"});
+            Action indexableObjectComparison = () => subject.ShouldBeEquivalentTo(expected);
+
+            indexableObjectComparison.ShouldThrow<NotSupportedException>();
```

Comment:
Could you add `.Message.ShouldBe(...)` on the end of this statement? It helps present a full picture of the user experience in the test code, and on rare occasions it has saved me from accidentally testing the wrong failure mode when the method throws the same exception type for two different reasons.